akka-docs/src/main/paradox/stream/operators/Source-or-Flow/ask.md

```diff
@@ -15,7 +15,7 @@ Use the `ask` pattern to send a request-reply message to the target `ref` actor.
 Use the `ask` pattern to send a request-reply message to the target `ref` actor.
 If any of the asks times out it will fail the stream with a @apidoc[AskTimeoutException].
 
-The `mapTo` class parameter is used to cast the incoming responses to the expected response type.
+The `S` generic parameter is used to cast the responses from the actor to the expected outgoing flow type.
```

Comment:
"If a different type of answer is sent this will lead to the future being failed with a class cast exception" or something like that?

Reply:
Good one - I wonder if we should throw a custom `ClassCastException` making it more clear where it's coming from

Reply:
I think that is a good idea 👍

Reply:
the Scaladoc also mention "Do not forget to include the expected response type...", which can be good to have here as well (for `@scala`)

Reply:
the text should be different for `@scala` and `@java`. for java there is `mapTo` parameter